src/DeserializerFactory.php

```diff
@@ -34,9 +35,25 @@ class DeserializerFactory {
 	 */
 	private $currentFactory;
 
-	public function __construct( Deserializer $dataValueDeserializer, EntityIdParser $idParser ) {
+	/**
+	 * @var DispatchableDeserializer|null $currentEntityDeserializer
+	 */
+	private $currentEntityDeserializer;
+
+	/**
+	 * @param Deserializer $dataValueDeserializer
+	 * @param EntityIdParser $idParser
+	 * @param DispatchableDeserializer|null $currentEntityDeserializer used instead of constructing
+	 *        a new current Deserializer for entities using a current DeserializerFactory.
+	 */
+	public function __construct(
+		Deserializer $dataValueDeserializer,
+		EntityIdParser $idParser,
+		DispatchableDeserializer $currentEntityDeserializer = null
+	) {
 		$this->legacyFactory = new LegacyDeserializerFactory( $dataValueDeserializer, $idParser );
 		$this->currentFactory = new CurrentDeserializerFactory( $dataValueDeserializer, $idParser );
```

Comment:
Huh, right. Looks like newStatementDeserializer should die though: it's only used (via WikibaseRepo::getInternalStatementDeserializer and EntityChange::getStatementDeserializer) to deserialize change rows that contain old style serialization of statements. We needed that fro a transition period for about a week a few years ago. I think it can die now (along with WikibaseRepo::getInternalStatementDeserializer and EntityChange::getStatementDeserializer).